File: datasets/Python/Copilot/Algorithm/easy5.py
def gameOfStones(n):
    dp = [False] * (n + 1)
    for i in range(1, n + 1):
        if i >= 2 and not dp[i - 2]:
            dp[i] = True
        elif i >= 3 and not dp[i - 3]:
            dp[i] = True
        elif i >= 5 and not dp[i - 5]:
            dp[i] = True
    return "First" if dp[n] else "Second"

# Manual Input
t = int(input("Enter number of test cases: "))
for _ in range(t):
    n = int(input("Enter number of stones: "))
    print(gameOfStones(n))

File: datasets/C++/Copilot/Algorithm/easy5.cpp
#include <iostream>
#include <vector>
using namespace std;

string gameOfStones(int n) {
    vector<bool> dp(n + 1, false);
    for (int i = 1; i <= n; ++i) {
        if (i >= 2 && !dp[i - 2]) dp[i] = true;
        else if (i >= 3 && !dp[i - 3]) dp[i] = true;
        else if (i >= 5 && !dp[i - 5]) dp[i] = true;
    }
    return dp[n] ? "First" : "Second";
}

int main() {
    int t, n;
    cout << "Enter number of test cases: ";
    cin >> t;
    while (t--) {
        cout << "Enter number of stones: ";
        cin >> n;
        cout << gameOfStones(n) << endl;
    }
    return 0;
}
